Automattic/wp-calypso
# Site Migration: Add upgrade component to site migration flow

Closes https://github.com/Automattic/wp-calypso/issues/87605

## Proposed Changes

Adds an upgrade plan screen in the migration flow. This is not a WIP anymore, I think we should merge this and work on email verification in a new PR, this is the issue tracking that: https://github.com/Automattic/wp-calypso/issues/87735.

## Testing Instructions



* Apply this PR or use the calypso.live link
* Create a new site and grab the slug
* Navigate to `/setup/site-migration/site-migration-upgrade-plan?flags=onboarding/new-migration-flow&siteSlug=[SITE_SLUG]`
* Click on `Continue`
* Buy a plan
* Verify you end up in `setup/site-migration/site-migration-instructions?flags=onboarding/new-migration-flow&siteSlug=[SITE_SLUG]`

## Pre-merge Checklist
